pulsar/producer_partition.go

```diff
@@ -126,6 +124,8 @@ type partitionProducer struct {
 	ctx                  context.Context
 	cancelFunc           context.CancelFunc
 	backOffPolicyFunc    func() backoff.Policy
+
+	buffersPool internal.BuffersPool
```

Comment:
This should be at the process level. It doesn't make sense to maintain the pool for each pulsar client.

I think maybe we don't need to add the custom labels for this process level metric.